### ISSUE TITLE
Unexpected Error for RTRIM Function in Where Clause

### ISSUE THREAD

**Issue by bajinsheng** [duckdb/duckdb#3616]
#### What happens?
An internal error bubbles up when using RTRIM function in the Where Clause.

#### To Reproduce
```sql
CREATE TABLE t0(c0 INT);
SELECT * FROM t0 WHERE RTRIM(false) > c0 OR 0 <= c0; 
-- Error: INTERNAL Error: Attempting to initialize state of expression of unknown type!
```

#### Environment (please complete the following information):
 - OS: Ubuntu 20.04.3 LTS
 - DuckDB Version: v0.3.5-dev359 35411c758
 - DuckDB Client: CLI

Found on DuckDB v0.3.5 by @kokrui @mrigger on experimental SQLancer features.

#### Before Submitting

- [x] **Have you tried this on the latest `master` branch?**
* **Python**: `pip install duckdb --upgrade --pre`
* **R**: `install.packages("https://github.com/duckdb/duckdb/releases/download/master-builds/duckdb_r_src.tar.gz", repos = NULL)`
* **Other Platforms**: You can find binaries [here](https://github.com/duckdb/duckdb/releases/tag/master-builds) or compile from source.

- [x] **Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?**

**Comment by hannes** [comment]
Confirmed, its a bug

**Comment by Mytherin** [comment]
Thanks for the report!

This seems to have been resolved by #3675, and was also related to the OR-clause pushdown.
